scripts/experiment_2/nbconverted/3_add_batch_effects.py

```diff
@@ -0,0 +1,137 @@
+
+
+# In[6]:
+
+
+# Read in UMAP model
+infile = open(umap_model_file, 'rb')
+umap_model = pickle.load(infile)
+infile.close()
+
+
+# In[7]:
+
+
+# Read in data
+simulated_data = pd.read_table(
+    simulated_data_file,
+    header=0, 
+    index_col=0,
+    compression='xz',
+    sep='\t')
+
+simulated_data.head(10)
+
+
+# In[8]:
+
+
+get_ipython().run_cell_magic('time', '', '# Add batch effects\nnum_simulated_samples = simulated_data.shape[0]\nnum_genes = simulated_data.shape[1]\n\n# Create an array of the simulated data indices\nsimulated_ind = np.array(simulated_data.index)\n\nfor i in num_batches:\n    print(\'Creating simulated data with {} batches..\'.format(i))\n    \n    batch_file = os.path.join(\n            base_dir,\n            "data",\n            "batch_simulated",\n            analysis_name,\n            "Batch_"+str(i)+".txt.xz")\n    \n    num_samples_per_batch = int(num_simulated_samples/i)\n    \n    if i == 1:        \n        simulated_data.to_csv(batch_file, sep=\'\\t\', compression=\'xz\')\n        \n    else:  \n        batch_data = simulated_data.copy()\n        \n        # Shuffle indices\n        np.random.shuffle(simulated_ind)\n        \n        for j in range(i):\n            #print(j)\n            \n            # Partition indices to batch\n            partition = np.array_split(simulated_ind, i)\n            \n            #print("before")\n            #print(batch_data.loc[partition[j].tolist()].head())\n            \n            #print("indices to change: {}".format(partition))\n            \n            # Scalar to shift gene expressiond data\n            stretch_factor = np.random.normal(0.0, 0.2, [1,num_genes])\n            \n            #print(stretch_factor)\n            \n            # Tile stretch_factor to be able to add to batches\n            num_samples_per_batch = len(partition[j])\n            stretch_factor_tile = pd.DataFrame(\n                pd.np.tile(\n                    stretch_factor,\n                    (num_samples_per_batch, 1)),\n                index=batch_data.loc[partition[j].tolist()].index,\n                columns=batch_data.loc[partition[j].tolist()].columns)\n            \n            #print(stretch_factor_tile.head())\n            \n            # Add batch effects\n            batch_data.loc[partition[j].tolist()] = batch_data.loc[partition[j].tolist()] + stretch_factor_tile\n            \n            #print("after")\n            #print(batch_data.loc[partition[j].tolist()].head())\n\n\n        # Should we re-normalize from 0-1 range?\n        #from sklearn import preprocessing\n        #batch_data = preprocessing.MinMaxScaler().fit_transform(batch_data)\n        #batch_data_df = pd.DataFrame(batch_data,\n        #                        columns=batch_data.columns,\n        #                        index=batch_data.index)\n            \n        #print(batch_data)\n            \n        # Save\n        batch_data.to_csv(batch_file, sep=\'\\t\', compression=\'xz\')')
```

Comment:
Looks like its an environment issue using ```%%time```.  This will not be an issue when I refactor the scripts so for now I will leave it as is.  But in the future that is good to know about.
https://github.com/jupyter/nbconvert/issues/975